### PR TITLE
Release | Fixed issued reported by SonarQube 

### PR DESCRIPTION
**Pull request by rene-ye**

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java [hunk 1]
```diff
@@ -6202,7 +6202,7 @@ void writeRPCReaderUnicode(String sName, Reader re, long reLength, boolean bOut,
 
     void sendEnclavePackage(String sql, ArrayList<byte[]> enclaveCEKs) throws SQLServerException {
         if (null != con && con.isAEv2()) {
-            if (null != sql && "" != sql && null != enclaveCEKs && 0 < enclaveCEKs.size() && con.enclaveEstablished()) {
+            if (null != sql && "".equals(sql) && null != enclaveCEKs && 0 < enclaveCEKs.size() && con.enclaveEstablished()) {
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: shouldn't this be !"".equals(sql)?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java [hunk 2]
```diff
@@ -139,33 +140,27 @@ private VSMAttestationResponse validateAttestationResponse(VSMAttestationRespons
...
+                }
+                processSDPEv1(userSql, preparedTypeDefinitions, params, parameterNames, connection, stmt, rs,
+                        enclaveRequestedCEKs);
+                // Process the third resultset.
+                if (connection.isAEv2() && stmt.getMoreResults()) {
+                    try (ResultSet hgsRs = (SQLServerResultSet) stmt.getResultSet()) {
+                        if (hgsRs.next()) {
+                            hgsResponse = new VSMAttestationResponse(hgsRs.getBytes(1));
+                            // This validates and establishes the enclave session if valid
+                            validateAttestationResponse();
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java]: `!connection.enclaveEstablished()` is removed, is that intentional?

File: src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java [hunk 3]
```diff
@@ -6202,7 +6202,7 @@ void writeRPCReaderUnicode(String sName, Reader re, long reLength, boolean bOut,
 
     void sendEnclavePackage(String sql, ArrayList<byte[]> enclaveCEKs) throws SQLServerException {
         if (null != con && con.isAEv2()) {
-            if (null != sql && "" != sql && null != enclaveCEKs && 0 < enclaveCEKs.size() && con.enclaveEstablished()) {
+            if (null != sql && !"".equals(sql) && null != enclaveCEKs && 0 < enclaveCEKs.size() && con.enclaveEstablished()) {
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java]: looks funny, use isEmpty instead?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 4]
```diff
@@ -1047,9 +1047,9 @@ private ResultSet executeSPFkeys(String[] procParams) throws SQLException, SQLTi
                 cstmt.setString(i + 1, procParams[i]);
             }
             String currentDB = null;
-            if (null != procParams[2] && procParams[2] != "") {// pktable_qualifier
+            if (null != procParams[2] && !"".equals(procParams[2])) {// pktable_qualifier
                 currentDB = switchCatalogs(procParams[2]);
-            } else if (null != procParams[5] && procParams[5] != "") {// fktable_qualifier
+            } else if (null != procParams[5] && !"".equals(procParams[5])) {// fktable_qualifier
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: same - isEmpty?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java [hunk 5]
```diff
@@ -139,33 +140,27 @@ private VSMAttestationResponse validateAttestationResponse(VSMAttestationRespons
...
+                    // No results. Meaning no parameter.
+                    // Should never happen.
+                    return enclaveRequestedCEKs;
+                }
+                processSDPEv1(userSql, preparedTypeDefinitions, params, parameterNames, connection, stmt, rs,
+                        enclaveRequestedCEKs);
+                // Process the third resultset.
+                if (connection.isAEv2() && stmt.getMoreResults()) {
+                    try (ResultSet hgsRs = (SQLServerResultSet) stmt.getResultSet()) {
+                        if (hgsRs.next()) {
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java]: Is it possible that the resultset is empty here? If it is, should we throw an exception or it is a legit scenario?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerAASEnclaveProvider.java [hunk 6]
```diff
@@ -102,47 +102,42 @@ public EnclaveSession getEnclaveSession() {
...
+                }
+                processSDPEv1(userSql, preparedTypeDefinitions, params, parameterNames, connection, stmt, rs,
+                        enclaveRequestedCEKs);
+                // Process the third resultset.
+                if (connection.isAEv2() && stmt.getMoreResults()) {
+                    try (ResultSet hgsRs = (SQLServerResultSet) stmt.getResultSet()) {
+                        if (hgsRs.next()) {
+                            hgsResponse = new AASAttestationResponse(hgsRs.getBytes(1));
+                            // This validates and establishes the enclave session if valid
+                            validateAttestationResponse();
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerAASEnclaveProvider.java]: The questions for VSM provider applies here too.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java (hunk 5 above)
Comment (rene-ye) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java]: It shouldn't happen but something can go wrong, we can throw an exception.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java (hunk 2 above)
Comment (rene-ye) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java]: Intentional.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerAASEnclaveProvider.java (hunk 6 above)
Comment (rene-ye) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerAASEnclaveProvider.java]: Intentional.

**Pull request merged**